buildalias/main.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/pelletier/go-toml/v2"
+	"log"
+	"net/http"
+	"os"
+	"sort"
+)
+
+type GameServerBuild struct {
+	BuildId string
+	Weight int
+}
+
+func main() {
+	args := os.Args
+	fileName := "thundernetes-buildalias.toml"
+	helpMessage := "\n\tThundernetes build alias feature\n\n" +
+	"1) create - generates a new build alias with the name provided." +
+	" Each alias is made of one or more pairs of build id and weights (priority of assignment for allocation). Template for using:" +
+	"\n\n\tbuildalias create <alias-name> <buildId1> <weight1> <buildId2> <weight2> ... <buildIdN> <weightN>\n\n" +
+	"2) allocate - create a new game server instance for the build alias and session id provided. Template for using\n\n\t" +
+	"buildalias allocate <alias-name> <sessionId>\n\n" +
+	"3) help - Displays this message and exit"
+
+	if len(args) == 1 {
+		fmt.Println(helpMessage)
+		return;
+	}
+
+	switch(args[1]) {
+	case "allocate":
+		AllocateForBuildAlias(args, fileName)
+
+	case "help":
+		fmt.Println(helpMessage)
+
+	default:
+		fmt.Println("Sorry, but the command "+ args[1] + " is not recognized.")
+	}
+}
+
+func AllocateForBuildAlias(args []string, fileName string) {
+	if len(args) < 6 {
+		log.Fatal("Not enough arguments were provided.");
+	}
+
+	buildName := args[2]
+	ipToAllocate := args[3]
+	portToAllocate := args[4]
+	sessionId := args[5]
+
+	idx := 0
+	aliasFound := false
+
+	fmt.Println("Allocating a Game Server for " + buildName)
+	ipToAllocate = "http://" + ipToAllocate + ":" + portToAllocate + "/api/v1/allocate"
+
+	aliasMap := make(map[string][]map[string][]map[string]interface{})

```

Comment:
any chance we can make it a strongly typed struct to avoid future issues? Check the example on the library you use https://github.com/pelletier/go-toml#getting-started